cmd/tsbs_load_cratedb/creator.go

```diff
@@ -0,0 +1,219 @@
+	if err != nil {
+		fatal("cannot parse the header: %v", err)
+		panic(err)
+	}
+	d.tableDefs = tableDefs
+
+	conn, err := pgx.Connect(*d.cfg)
+	if err != nil {
+		fatal("Cannot establish a connection to database: %v", err)
+		panic(err)
+	}
+	d.conn = conn
+}
+
+// readDataHeader fills the dbCreator struct with the data structure
+// (tables description) specified at the beginning of the data file.
+//
+// First N lines are the header, describing the data structure.
+// The first line contains the tags table name followed by a comma-separated
+// list of tags:
+//     tags,hostname,region,datacenter,rack,os,arch,team,service,service_version
+//
+// The second through N-1 line containing table name (ex.: 'disk') followed by
+// list of column names, comma-separated:
+//     disk,total,free,used,used_percent,inodes_total,inodes_free,inodes_used
+//
+// The last line being blank to separate the header from the data.
+//
+// Header example:
+//      tags,hostname,region,datacenter,rack,os,arch,team,service,service_version,service_environment
+//      disk,total,free,used,used_percent,inodes_total,inodes_free,inodes_used
+//      nginx,accepts,active,handled,reading,requests,waiting,writing
+func (d *dbCreator) readDataHeader(br *bufio.Reader) ([]*tableDef, error) {
+	var tableDefs []*tableDef
+
+	line, err := br.ReadString('\n')
+	if err != nil {
+		return nil, err
+	}
+	line = strings.TrimSpace(line)
+	tagsLine := strings.Split(line, ",")
+	if tagsLine[0] != "tags" {
+		return nil, errors.New("first header line doesn't contain tags")
+	}
+	tags := tagsLine[1:]
+
+	for {
+		line, err := br.ReadString('\n')
+		if err != nil {
+			return nil, err
+		}
+		line = strings.TrimSpace(line)
+		if len(line) == 0 {
+			break
+		}
+
+		parts := strings.SplitN(line, ",", 2)
+		if len(parts) < 2 {
+			return nil, errors.New("metric columns are missing")
+		}
+		tableDefs = append(
+			tableDefs,
+			&tableDef{
+				name: parts[0],
+				tags: tags,
+				cols: strings.Split(parts[1], ","),
+			},
+		)
+	}
+	return tableDefs, nil
+}
+
+func (d *dbCreator) CreateDB(dbName string) error {
+	for _, tableDef := range d.tableDefs {
+		// the dbName(schema) is required by the load.Processor implementation,
+		// therefore, it is assigned to a table definition for the further usage
+		tableDef.schema = dbName
+		err := d.createMetricsTable(tableDef)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (d *dbCreator) createMetricsTable(table *tableDef) error {
+	var tagsObjectChildCols []string
+	for _, column := range table.tags {
+		tagsObjectChildCols = append(
+			tagsObjectChildCols,
+			fmt.Sprintf("%s %s", column, "string"))
+	}
+
+	var metricCols []string
+	for _, column := range table.cols {
+		metricCols = append(
+			metricCols,
+			fmt.Sprintf("%s %s", column, "double"))
+	}
+
+	// TODO partition table by configurable time interval
+	sql := fmt.Sprintf(`
+		CREATE TABLE %s (
+			tags object as (%s),
+			ts timestamptz,
```

Comment:
This does not appear to work with crate 3.3.2, I get:
```
SQLParseException: Cannot find data type: timestamptz (SQLSTATE XX000)
```

Reply:
Thank you for reviewing the PR!

I've been using a nightly build for development that contains some new date time types and aliases. I've pushed a fixup, i think in this case it is ok to use the `timestamp` data type, it has the `time with time zone` and will remain as it is for a couple of future releases.